grass/kernel.c

```diff
@@ -48,9 +48,19 @@ void excp_entry(uint id) {
 static void proc_yield();
 static void proc_syscall();
 
-uint proc_curr_idx;
+uint proc_curr_idx, waiting = 0, wakeup = 0;
 struct process proc_set[MAX_NPROCESS];
 
+struct kernel_msg
+{
+    int in_use;
+    int sender;
+    int receiver;
+    char msg[SYSCALL_MSG_LEN];
+};
+
+struct kernel_msg *KERNEL_MSG_BUFF;
```

Comment:
Let's use lowercase for variable names. I feel like we can call the type `struct pending_ipc` and then define the variable `struct pending_ipc *pending_ipc_buffer`.